app/models/internal/impl/PagerDutyNotificationEntry.java

```diff
@@ -137,33 +138,110 @@ public Builder() {
         }
     }
 
+    /**
+     * Model classes for a PagerDuty alert and the embedded context objects, as defined in their API document:
+     * https://v2.developer.pagerduty.com/docs/events-api
+     */
+    private class PagerDutyAlert {
```

Comment:
Can you use the Builder pattern here?  We prefer immutable objects whenever we can.